projects/Funkwhale/services/funkwhale/module.nix

```diff
@@ -0,0 +1,558 @@
+      default = null;
+      example = "/run/secrets/funkwhale.env";
+      description = ''
+        Secret environment variable settings. These overwrite `services.funkwhale.settings`.
+        See <https://docs.funkwhale.audio/administrator/configuration/env-file.html>.
+
+        Example:
+
+        ```
+        DJANGO_SECRET_KEY="some-very-increadibly-secret-key"
+        TYPESENSE_API_KEY="some-other-very-increadibly-super-secret-key"
+        ```
+
+        `DJANGO_SECRET_KEY` will be auto-generated if not set here.
+      '';
+    };
+  };
+
+  config = lib.mkIf cfg.enable {
+    assertions = [
+      {
+        assertion = !(cfg.settings ? DJANGO_SECRET_KEY);
+        message = ''
+          Setting `services.funkwhale.settings.DJANGO_SECRET_KEY` is not secure.
+          Instead, `DJANGO_SECRET_KEY` should be set in `services.funkwhale.secretEnvironmentFile`.
+        '';
+      }
+    ];
+
+    services.funkwhale.settings = lib.mkMerge [
+      {
+        MEDIA_ROOT = "/var/lib/funkwhale/data/media";
+        MUSIC_DIRECTORY_PATH = "/var/lib/funkwhale/data/music";
+      }
+      (lib.mkIf cfg.createDatabaseLocally {
+        DATABASE_HOST = "";
+        DATABASE_PORT = 5432;
+        DATABASE_USER = cfg.user;
+        # Peer authentication.
+        DATABASE_PASSWORD = "";
+      })
+      (lib.mkIf cfg.createRedisLocally {
+        CACHE_URL = "unix://${config.services.redis.servers.funkwhale.unixSocket}?db=0";
+        CELERY_BROKER_URL = "redis+socket://${config.services.redis.servers.funkwhale.unixSocket}?virtual_host=0";
+      })
+      (lib.mkIf cfg.configureNginx {
+        REVERSE_PROXY_TYPE = "nginx";
+        NGINX_MAX_BODY_SIZE = lib.mkDefault "100M";
+      })
+    ];
+
+    users.users = lib.mkIf (cfg.user == "funkwhale") {
+      ${cfg.user} = {
+        isSystemUser = true;
+        group = cfg.group;
+      };
+    };
+    users.groups = lib.mkIf (cfg.group == "funkwhale") {
+      ${cfg.group} = { };
+    };
+
+    environment.systemPackages = [
+      (pkgs.writeShellScriptBin "funkwhale-manage" ''
+        if [[ ! "$USER" == "${cfg.user}" ]]; then
+          echo "This has to be run with the \`${cfg.user}\` user. Ex: \`sudo -u ${cfg.user} funkwhale-manage\`"
+          exit 1
+        fi
+
+        # `troi.utils.discover_patches()` will error, because it looks for patches
+        # in the current directory.
+        stat . &> /dev/null || echo -e \
+          "WARNING: funkwhale-manage should be run from a directory the ${cfg.user} user has read access to.\n"
+
+        # Set `cfg.settings` environment variables.
+        ${lib.concatMapAttrsStringSep "\n" (
+          name: value: ''export ${name}=${lib.escapeShellArg value}''
+        ) cfg.settings}
+
+        # Set secret environment variables.
+        set -o allexport
+        if [ -f "/var/lib/funkwhale/config/django_secret_key.env" ]; then
+          source "/var/lib/funkwhale/config/django_secret_key.env"
+        fi
```

Comment:
`/var/lib/funkwhale/config/django_secret_key.env` shouldn't be sourced when `cfg.secretEnvironmentFile` is set.